filters/auth/tokeninfocache.go

```diff
@@ -1,48 +1,55 @@
 package auth
 
 import (
-	"container/list"
+	"maps"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/metrics"
 )
 
 type (
 	tokeninfoCache struct {
-		client tokeninfoClient
-		size   int
-		ttl    time.Duration
-		now    func() time.Time
-
-		mu    sync.Mutex
-		cache map[string]*entry
-		// least recently used token at the end
-		history *list.List
+		client  tokeninfoClient
+		metrics metrics.Metrics
+		size    int
+		ttl     time.Duration
+		now     func() time.Time
+
+		cache sync.Map     // map[string]*entry
+		count atomic.Int64 // estimated number of cached entries, see https://github.com/golang/go/issues/20680
+		done  chan struct{}
```

Comment:
done -> quit? 
I think we most often use quit.

Reply:
Right, will fix
```
ayastrebov@zamd:~/src/github.com/zalando/skipper$ grep -rn -E 'done\s*chan ' | wc -l
2
ayastrebov@zamd:~/src/github.com/zalando/skipper$ grep -rn -E 'quit\s*chan ' | wc -l
22
```